dm/syncer/sharding_group.go

```diff
@@ -239,7 +239,8 @@ func (sg *ShardingGroup) CheckSyncing(source string, location binlog.Location) (
 	if activeDDLItem == nil {
 		return true
 	}
-	return binlog.CompareLocation(activeDDLItem.FirstLocation, location, sg.enableGTID) > 0
+	// ddl position caculate use last_position's gtid, when last is a dml, it will be equal, but the dml should be synced.
+	return binlog.CompareLocation(activeDDLItem.FirstLocation, location, sg.enableGTID) >= 0
```

Comment:
I don't think this is a solution. The issue is because last_position is wrong after https://github.com/pingcap/dm/pull/1926, the correct behavior is 
```
DML 
# activeDDLItem.FirstLocation  <-- (endlocation of DML, start location of DDL, lastLocation)
DDL
```
after https://github.com/pingcap/dm/pull/1926
```
DML
DDL
activeDDLItem.FirstLocation(gtid) <-- (endlocation of DDL, lastLocation)
```
so now binlog.CompareLocation(activeDDLItem.FirstLocation, location, sg.enableGTID) greater than 0

Reply:
But I forgot why we update last location when receive begin, cc @lance6716

Reply:
maybe to some problems of exit safe point. I'll take a look later